plugins/obs-qsv11/QSV_Encoder_Internal.cpp

```diff
@@ -333,6 +367,64 @@ bool QSV_Encoder_Internal::InitParams(qsv_param_t *pParams)
 	return true;
 }
 
+bool QSV_Encoder_Internal::InitParamsAV1(qsv_param_t *pParams)
+{
+	memset(&m_mfxEncParams, 0, sizeof(m_mfxEncParams));
+
+	m_mfxEncParams.mfx.CodecId = MFX_CODEC_AV1;
+	m_mfxEncParams.mfx.GopOptFlag = MFX_GOP_STRICT;
+	m_mfxEncParams.mfx.NumSlice = 1;
+	m_mfxEncParams.mfx.TargetUsage = pParams->nTargetUsage;
+	m_mfxEncParams.mfx.CodecProfile = pParams->nCodecProfile;
+	m_mfxEncParams.mfx.FrameInfo.FrameRateExtN = pParams->nFpsNum;
+	m_mfxEncParams.mfx.FrameInfo.FrameRateExtD = pParams->nFpsDen;
+	m_mfxEncParams.mfx.FrameInfo.FourCC = MFX_FOURCC_NV12;
+	m_mfxEncParams.mfx.FrameInfo.ChromaFormat = MFX_CHROMAFORMAT_YUV420;
+	m_mfxEncParams.mfx.FrameInfo.PicStruct = MFX_PICSTRUCT_PROGRESSIVE;
+	m_mfxEncParams.mfx.FrameInfo.CropX = 0;
+	m_mfxEncParams.mfx.FrameInfo.CropY = 0;
+	m_mfxEncParams.mfx.FrameInfo.CropW = pParams->nWidth;
+	m_mfxEncParams.mfx.FrameInfo.CropH = pParams->nHeight;
+	m_mfxEncParams.mfx.GopRefDist = 1;
+
+	m_mfxEncParams.mfx.RateControlMethod = pParams->nRateControl;
+
+	m_mfxEncParams.AsyncDepth = pParams->nAsyncDepth;
+
+	switch (pParams->nRateControl) {
+	case MFX_RATECONTROL_CBR:
+		m_mfxEncParams.mfx.TargetKbps = pParams->nTargetBitRate;
+		m_mfxEncParams.mfx.BufferSizeInKB = pParams->nTargetBitRate * 2;
+		m_mfxEncParams.mfx.InitialDelayInKB =
+			pParams->nTargetBitRate * 1;
+		break;
+	case MFX_RATECONTROL_VBR:
+		m_mfxEncParams.mfx.TargetKbps = pParams->nTargetBitRate;
+		m_mfxEncParams.mfx.MaxKbps = pParams->nMaxBitRate;
+		m_mfxEncParams.mfx.BufferSizeInKB = pParams->nTargetBitRate * 2;
+		m_mfxEncParams.mfx.InitialDelayInKB =
+			pParams->nTargetBitRate * 1;
+		break;
+	default:
+		break;
+	}
+
+	// Width must be a multiple of 16
+	// Height must be a multiple of 16 in case of frame picture and a
```

Comment:
Does this still apply here for AV1? And will the output have the correct resolution regardless?

Reply:
This does apply for AV1. Any resolution given will be correct as long as the MSDK_ALIGN16 is used.